ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -4020,7 +4151,7 @@ static vk_pipeline ggml_vk_guess_matmul_pipeline(ggml_backend_vk_context * ctx,
         return aligned ? mmp->a_s : mmp->s;
     }
 
-    if ((ctx->device->mul_mat_s[src0_type] && (m <= 32 || n <= 32)) || (!ctx->device->mul_mat_m[src0_type] && !ctx->device->mul_mat_l[src0_type])) {
+    if ((ctx->device->mul_mat_s[src0_type] && (m <= 32 || n <= 32)) || (!ctx->device->mul_mat_m[src0_type] && !ctx->device->mul_mat_l[src0_type]) || src1_type == GGML_TYPE_Q8_1) {
```

Comment:
Why does the int path force the small size?

Reply:
Just tests. Larger sizes don't get better performance, I assume due to shared memory getting overloaded with caching 32 k-vals for each of the input matrices. I haven't looked into ways of improving this to be able to increase block sizes yet. If you have tuning suggestions, let me know.

Reply:
I retested it with the finished implementation, now it behaves the same as fp16 (all sizes on Nvidia, small and medium on Intel and AMD). Performance increased by a lot for all of them with the larger tile sizes. Nvidia DP4A matmul now nearly matches coopmat1, AMD and Intel nearly doubled in speed. Nice!